src/main/java/com/hierynomus/smbj/share/NamedPipe.java

```diff
@@ -111,7 +120,48 @@ public int read(byte[] buffer, int offset, int length) {
      * @return the output message
      */
     public byte[] transact(byte[] inBuffer) {
-        return ioctl(FSCTL_PIPE_TRANSCEIVE, true, inBuffer, 0, inBuffer.length);
+        final SMB2IoctlResponse ioctlResponse = share.ioctl(
+            fileId,
+            FSCTL_PIPE_TRANSCEIVE,
+            true,
+            new ArrayByteChunkProvider(inBuffer, 0, inBuffer.length, 0),
+            -1,
+            SUCCESS_OR_BUFFER_OVERFLOW
+        );
+
+        final ByteArrayOutputStream outBuffer = new ByteArrayOutputStream(4096);
+        final byte[] ioctlOutputBuffer = ioctlResponse.getOutputBuffer();
+        try {
+            outBuffer.write(ioctlOutputBuffer);
+        } catch (final IOException exception) {
+            throw new SMBRuntimeException(exception);
+        }
+
+        final NtStatus ioctlStatus = ioctlResponse.getHeader().getStatus();
+        if (ioctlStatus.equals(NtStatus.STATUS_BUFFER_OVERFLOW)) {
+            for (;;) {
+                final SMB2ReadResponse readResponse = share.read(
+                    fileId,
+                    0,
+                    share.getReadBufferSize(),
+                    SUCCESS_OR_EOF_OR_BUFFER_OVERFLOW
+                );
+
+                final byte[] readData = readResponse.getData();
+                try {
+                    outBuffer.write(readData);
+                } catch (final IOException exception) {
+                    throw new SMBRuntimeException(exception);
+                }
+
+                final NtStatus readStatus = readResponse.getHeader().getStatus();
+                if (!readStatus.equals(NtStatus.STATUS_BUFFER_OVERFLOW)) {
```

Comment:
Also this will stop if the response is either EOF or SUCCESS which looks to be correct.  While I haven't seen a EOF, if we ever got one, than that's a good indicator that this code should stop as well.  :)